### PR TITLE
Make Scaladoc actually exit with non-zero exit code in case of errors, as its docs say it does

### PR DESCRIPTION
**Pull request by dturner-tw**
I did this work for Twitter, on Twitter's time, so I'm covered under Twitter's CLA. 

Fixes https://issues.scala-lang.org/browse/SI-8885.

### CODE REVIEW

File: src/scaladoc/scala/tools/nsc/ScalaDoc.scala [hunk 1]
```diff
@@ -50,12 +46,18 @@ class ScalaDoc {
...
+    !reporter.reallyHasErrors
   }
 }
 
+class ScalaDocReporter(settings: Settings) extends ConsoleReporter(settings) {
+
+  // need to do sometimes lie so that the Global instance doesn't
+  // trash all the symbols just because there was an error
+  override def hasErrors = false
+  def reallyHasErrors = super.hasErrors
```
Comment (som-snytt) [inline src/scaladoc/scala/tools/nsc/ScalaDoc.scala:58]: I'd retweet this if there were a button.  T/F is never enough.

**Comment by adriaanm** [comment]
Thanks & welcome, @dturner-tw! Unfortunately, we cannot accept contributions without a signed [CLA](http://typesafe.com/contribute/cla/scala).

**Comment by adriaanm** [comment]
(Note that I'm referring to the Scala CLA, not the Typesafe CLA.)

**Comment by dturner-tw** [comment]
OK, I signed the Scala CLA (with permission from Twitter's OSS people).

**Comment by adriaanm** [comment]
Thanks!

**Comment by adriaanm** [comment]
LGTM! I verified the doc task still works in sbt. @som-snytt, would you RT the LG?

```
resolvers += "pr" at "http://private-repo.typesafe.com/typesafe/scala-pr-validation-snapshots/"

scalaVersion := "2.11.3-5f29a26-SNAPSHOT"

> doc
[info] Main Scala API documentation to /Users/adriaan/Desktop/scratch/target/scala-2.11.3-5f29a26-SNAPSHOT/api...
model contains 2 documentable templates
[info] Main Scala API documentation successful.
```

**Comment by dturner-tw** [comment]
That was a delightful experience.  A++++, would submit again!

**Comment by gkossakowski** [comment]
Our pleasure!

**Pull request merged**